modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Copy.java

```diff
@@ -0,0 +1,145 @@
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeEnum;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.Attribute;
+import org.ballerinalang.natives.annotations.BallerinaAnnotation;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Copy Function
+ */
+@BallerinaFunction(
+        packageName = "ballerina.lang.files",
+        functionName = "copy",
+        args = {@Argument(name = "source", type = TypeEnum.STRUCT, structType = "File",
+                structPackage = "ballerina.lang.files"),
+                @Argument(name = "destination", type = TypeEnum.STRUCT, structType = "File",
+                        structPackage = "ballerina.lang.files")},
+        isPublic = true
+)
+@BallerinaAnnotation(annotationName = "Description", attributes = { @Attribute(name = "value",
+        value = "This function copies a file from a given location to another") })
+@BallerinaAnnotation(annotationName = "Param", attributes = { @Attribute(name = "source",
+        value = "File that should be copied") })
+@BallerinaAnnotation(annotationName = "Param", attributes = { @Attribute(name = "destination",
+        value = "The location where the File should be pasted") })
+public class Copy extends AbstractNativeFunction {
+
+    @Override public BValue[] execute(Context context) {
+
+        BStruct source = (BStruct) getArgument(context, 0);
+        BStruct destination = (BStruct) getArgument(context, 1);
+
+            File file = new File(source.getValue(0).stringValue());
+            File destinationFile = new File(destination.getValue(0).stringValue());
+
+            File parent = destinationFile.getParentFile();
+            if (parent != null) {
+                if (!parent.exists()) {
+                    if (!parent.mkdirs()) {
+                        throw new BallerinaException("Error in writing file");
+                    }
+                }
+            }
+            if (!copy(file, destinationFile)) {
+                throw new BallerinaException("Error while copying file");
+            }
+
+        return VOID_RETURN;
+    }
+
+    private boolean copy(File src, File dest) {
+
+        InputStream in = null;
+        OutputStream out = null;
+        try {
+
+            if (src.isDirectory()) {
+
+                if (!dest.exists()) {
+                    if (!dest.mkdir()) {
+                        return false;
+                    }
+                }
+
+                String files[] = src.list();
```

Comment:
I needed the String values of the paths to create a file object including the parent (line 101).